custom-widget-builder/test/HtmlTemplateGenerator.test.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2021 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {CustomWidgetBuilder} from "../src/CustomWidgetBuilder";
+import {HtmlTemplatesGenerator} from "../src/HtmlTemplatesGenerator";
+import * as os from "os";
+
+const fs = require('fs');
+
+describe('CustomWidgetBuilder', () => {
+
+  let builder: CustomWidgetBuilder;
+  let tempDir: string;
+
+  beforeAll(() => {
+    builder = new CustomWidgetBuilder();
+    tempDir = fs.mkdtempSync(os.tmpdir());
+  });
+
+  afterAll(() => {
+    fs.rmdirSync(tempDir, {recursive: true});
+  });
+
+  test('should generate correct html templates when a simple web component is given as input', async () => {
+    handleSimpleWC("WcExample.ts");
+  });
+
+  test('should generate correct html templates when a complex web component is given as input', async () => {
+    builder.generate("test/resources/pb-input.ts", tempDir);
```

Comment:
Be careful, in this test classes you will test html Generator AND json generation. 
Do you think it's better to test in this classes only HTML generation ?

Perhaps it could be nice to have a global test to verify each "step" is called for generation, and make unit test separately

Reply:
As you can see, the html template generator takes a PropertiesInfo in its contructor. So yes, we use the PropertiesInfo generated by the tool (which is not the PropertiesJsonGenerator).
So yes, we could build a PropertiesInfo "manually", but does it worth the effort ?